modules/programs/eww.nix

```
@@ -45,26 +96,60 @@ in {
     };
   };
 
-  config = mkIf cfg.enable {
-    home.packages = [ cfg.package ];
-    xdg.configFile."eww".source = cfg.configDir;
-
-    programs.bash.initExtra = mkIf cfg.enableBashIntegration ''
-      if [[ $TERM != "dumb" ]]; then
-        eval "$(${ewwCmd} shell-completions --shell bash)"
-      fi
-    '';
-
-    programs.zsh.initExtra = mkIf cfg.enableZshIntegration ''
-      if [[ $TERM != "dumb" ]]; then
-        eval "$(${ewwCmd} shell-completions --shell zsh)"
-      fi
-    '';
-
-    programs.fish.interactiveShellInit = mkIf cfg.enableFishIntegration ''
-      if test "$TERM" != "dumb"
-        eval "$(${ewwCmd} shell-completions --shell fish)"
-      end
-    '';
-  };
+  config = mkIf cfg.enable (mkMerge [
+    {
+      assertions = [
+          {
+            assertion = !(cfg.scssConfig == null && cfg.configDir == null) || !(cfg.yuckConfig == null && cfg.configDir == null);
+            message = "You can have either yuck & scss or configDir enabled";
+          }
+        ];
+
+      home.packages = [ cfg.package ];
+
+      programs.bash.initExtra = mkIf cfg.enableBashIntegration ''
+        if [[ $TERM != "dumb" ]]; then
+          eval "$(${ewwCmd} shell-completions --shell bash)"
+        fi
+      '';
+
+      programs.zsh.initExtra = mkIf cfg.enableZshIntegration ''
+        if [[ $TERM != "dumb" ]]; then
+          eval "$(${ewwCmd} shell-completions --shell zsh)"
+        fi
+      '';
+
+      programs.fish.interactiveShellInit = mkIf cfg.enableFishIntegration ''
+        if test "$TERM" != "dumb"
+          eval "$(${ewwCmd} shell-completions --shell fish)"
+        end
+      '';
+    }
+
+    (mkIf cfg.configDir
+      == null { xdg.configFile."eww".source = cfg.configDir; })
+
+    (mkIf cfg.yuckConfig == null { xdg.configFile."eww/eww.yuck".text = cfg.yuckConfig; })
+
+    (mkIf cfg.scssConfig == null { xdg.configFile."eww/eww.scss".text = cfg.scssConfig; })
+
+    (mkIf cfg.systemd.enable {
+      systemd.user.services.eww = {
+        Unit = {
+          Description = "ElKowars wacky widgets daemon";
+          Documentation = "https://elkowar.github.io/eww/";
+          After = [ "graphical-session.target" ];
+          PartOf = [ "graphical-session.target" ];
+        };
+
+        Service = {
+          ExecStart = "${cfg.package} daemon --no-daemonize";
+          ExecStop = "${cfg.package} kill";
+          ExecReload = "${cfg.package} reload";
```

```suggestion
          ExecStart = "${ewwCmd} daemon --no-daemonize";
          ExecStop = "${ewwCmd} kill";
          ExecReload = "${ewwCmd} reload";
```
`${cfg.package}` is just the Nix store path for Eww, not the `eww` executable.

---

ops my bad